module/mempool/consensus/exec_fork_suppressor.go

```diff
@@ -171,15 +147,45 @@ func (s *ExecForkSuppressor) Add(newSeal *flow.IncorporatedResultSeal) (bool, er
 // All returns all the IncorporatedResultSeals in the mempool
 func (s *ExecForkSuppressor) All() []*flow.IncorporatedResultSeal {
 	s.mutex.RLock()
-	defer s.mutex.RUnlock()
-	return s.seals.All()
+	seals := s.seals.All()
+	s.mutex.RUnlock()
+	sealsByBlockID := make(map[flow.Identifier]sealsList, 0)
+	for _, seal := range seals {
+		sealsPerBlock := sealsByBlockID[seal.Seal.BlockID]
+		sealsByBlockID[seal.Seal.BlockID] = append(sealsPerBlock, seal)
+	}
+
+	return s.filterConflictingSeals(sealsByBlockID)
```

Comment:
Who would call seal mempool's All method? If that's not builder, is it OK to just return all the seals even if there are conflictions?

Reply:
`All` is not used at all, but I believe it should perform needed checks since this mempool tracks execution forks. 
We could rebuild builder for using `All` instead of `ByID`, but definitely not now.